pandas/tests/reshape/test_concat.py

```diff
@@ -1110,6 +1110,23 @@ def test_append_empty_frame_to_series_with_dateutil_tz(self):
         result = df.append([s, s], ignore_index=True)
         tm.assert_frame_equal(result, expected)
 
+    def test_append_empty_tz_frame_with_datetime64ns(self):
+        # https://github.com/pandas-dev/pandas/issues/35460
+        df = pd.DataFrame(columns=["a"]).astype("datetime64[ns, UTC]")
+
+        # pd.NaT gets inferred as tz-naive, so append result is tz-naive
```

Comment:
is it obvious this is the correct behavior?  id expect object dtype result

Reply:
Ah, good point. Yes, it should indeed give object dtype (at least that's how we concat tz-naive/tz-aware otherwise).

Will need to check why it becomes datetime64 here ..

Reply:
Yeah, so the reason is that we drop the empty arrays in `concat_compat`:

https://github.com/pandas-dev/pandas/blob/1a21836a9ade84cc95fca9e0dce42eb495b36384/pandas/core/dtypes/concat.py#L139-L141

so the empty tz-aware part is dropped, and the concat further only sees the tz-naive NaT.

Reply:
@jbrockmendel thoughts on this? This PR is for sure fixing an AssertionError that otherwise bubbled to the user (and which was a regression), which should not happen, but not fully sure what we should do with the expected result.